src/debug.h

```diff
@@ -117,8 +117,8 @@ const char *GetDebugString();
 }
 
 
-void ShowInfo(const char *str);
-void CDECL ShowInfoF(const char *str, ...) WARN_FORMAT(1, 2);
+void ShowInfo(const std::string &str);
+#define ShowInfoF(format_string, ...) ShowInfo(fmt::format(FMT_STRING(format_string), ## __VA_ARGS__))
```

Comment:
The biggest caveat with templating that function is also losing `FMT_STRING`, which brings us compile time validation of the format string: https://fmt.dev/latest/api.html#compile-time-format-string-checks.

Reply:
True, though if I'm reading the docs correctly we get it for free with C++20. Might be a way off though

Reply:
Yes, with C++20 you do not need the macro anymore. However, making it a template doesn't seem to do the trick: https://godbolt.org/z/r1Yvv4T9M
The macro version rightfully fails when trying to format a string as digit, but the template version fails because the format (string) isn't a constant expression even when the format is right.